src/gentypes.jl

```diff
@@ -477,17 +477,23 @@ function typecode(rosname::String, super::Symbol, members::Vector)
     #(4) Convert from PyObject
     push!(exprs, :(
         function convert(jlt::Type{$jlsym}, o::PyObject)
-            if convert(String, o["_type"]) != _typerepr(jlt)
+            if convert(String, o."_type") != _typerepr(jlt)
                 throw(InexactError(:convert, $jlsym, o))
             end
             jl = $jlsym()
             #Generated code here
             jl
         end
     ))
-    #(5) Accessing member variables through getindex
+    #(5) Accessing member variables through getproperty
     push!(exprs, :(
-        getindex(::Type{$jlsym}, s::Symbol) = RobotOS._rospy_objects[$rosname][s]
+        getproperty(::Type{$jlsym}, s::Symbol) = begin
+            try getproperty(RobotOS._rospy_objects[$rosname], s)
+            catch ex
+                isa(ex, KeyError) || rethrow(ex)
+                getfield($jlsym, s)
+            end
+        end
```

Comment:
Change to `function` definition style instead of a `begin` block?

Reply:
Good call, changed.